<?xml version="1.0"?>
<launch>

<arg name="meshfile" default="$(find uos_gazebo_worlds)/Media/models/avz_neu.dae" />
<arg name="device" default="cpu" />

<node pkg="rmcl" type="pcl2_to_scan" name="pcl2_to_scan" output="screen">
    <remap from="cloud" to="velodyne_points" />

    <rosparam file="$(find rmcl)/config/sensors/VLP-16-440.yaml" />
    <!-- UNCOMMENT FOR REAL VELODYNE -->
    <!-- <rosparam file="$(find mamcl)/config/sensors/VLP-16.yaml" /> -->
</node>

<!-- ICP with given pose estimate -->
<node pkg="rmcl" type="multi_lidar_corrector_embree" name="multi_lidar_corrector" 
    output="screen" if="$(eval arg('device') == 'cpu')">
    <param name="map_file" type="string" value="$(arg meshfile)" />
    <param name="odom_frame" type="string" value="odom_combined" />
    <param name="base_frame" type="string" value="base_footprint" />
    <remap from="scan" to="/pcl2_to_scan/scan"/>
    <remap from="pose" to="/move_base_simple/goal"/>
    <param name="max_distance" type="double" value="0.5" />
</node>

<node pkg="rmcl" type="multi_lidar_corrector_optix" name="multi_lidar_corrector" 
    output="screen" if="$(eval arg('device') == 'gpu')">
    <param name="map_file" type="string" value="$(arg meshfile)" />
    <param name="odom_frame" type="string" value="odom_combined" />
    <param name="base_frame" type="string" value="base_footprint" />
    <remap from="scan" to="/pcl2_to_scan/scan"/>
    <remap from="pose" to="/move_base_simple/goal"/>
    <param name="max_distance" type="double" value="0.5" />
    <param name="particles" type="int" value="5000" />
</node>

</launch>